test/functional/wallet_migration.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test Migrating a wallet from legacy to descriptor."""
+
+import os
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+)
+
+
+class WalletMigrationTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+        self.extra_args = [[]]
+        self.supports_cli = False
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_sqlite()
+        self.skip_if_no_bdb()
+
+    def assert_is_sqlite(self, wallet_name):
+        wallet_file_path = os.path.join(self.nodes[0].datadir, "regtest/wallets", wallet_name, self.wallet_data_filename)
+        with open(wallet_file_path, 'rb') as f:
+            file_magic = f.read(16)
+            assert_equal(file_magic, b'SQLite format 3\x00')
+        assert_equal(self.nodes[0].get_wallet_rpc(wallet_name).getwalletinfo()["format"], "sqlite")
+
+    def test_basic(self):
+        default = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+
+        self.log.info("Test migration of a basic keys only wallet without balance")
+        self.nodes[0].createwallet(wallet_name="basic0")
+        basic0 = self.nodes[0].get_wallet_rpc("basic0")
+        assert_equal(basic0.getwalletinfo()["descriptors"], False)
+
+        addr = basic0.getnewaddress()
+        change = basic0.getrawchangeaddress()
+
+        assert_equal(basic0.getaddressinfo(addr)["ismine"], True)
+        assert_equal(basic0.getaddressinfo(change)["ismine"], True)
+
+        basic0.migratewallet()
+        assert_equal(basic0.getwalletinfo()["descriptors"], True)
+        self.assert_is_sqlite("basic0")
+
+        addr_info = basic0.getaddressinfo(addr)
+        assert_equal(addr_info["ismine"], True)
+        assert_equal(basic0.getaddressinfo(change)["ismine"], True)
+        assert_equal(addr_info["hdkeypath"], "m/0'/0'/0'")
+
+        addr_info = basic0.getaddressinfo(basic0.getnewaddress("", "bech32"))
+        assert_equal(addr_info["hdkeypath"], "m/84'/1'/0'/0/0")
+
+        self.log.info("Test migration of a basic keys only wallet with a balance")
+        self.nodes[0].createwallet(wallet_name="basic1")
+        basic1 = self.nodes[0].get_wallet_rpc("basic1")
+        assert_equal(basic1.getwalletinfo()["descriptors"], False)
+
+        for i in range(0, 10):
+            default.sendtoaddress(basic1.getnewaddress(), 1)
```

Comment:
test_framework.authproxy.JSONRPCException: Requested wallet does not exist or is not loaded (-18)


https://cirrus-ci.com/task/6547488835371008?logs=ci#L4027